### ISSUE TITLE
Add support for persistent jobs (daemons)

### ISSUE THREAD

**Issue by Jacketbg** [jhuckaby/Cronicle#190]
First of all Cronicle is a great piece of software! Congrats!
There's only one small thing missing at least for my use case. Instead of periodic jobs I have lots of persistent processes. They should run 24/7 and in case of a crash, they need to be restarted (immediately or with some delay). Now I know many people will say "just use systemd", but with more than 20 such tasks per server it is very hard to manage them all and Cronicle looks like like the tool I need for this. It even kinda works right now by configuring the event with a `* * * * *` cron rule and setting it as a singleton, but this way my logs are becoming full of warnings for overruns. If those warnings could be suppressed on an event basis, this could be all it takes for this to work great :) Please consider.

**Comment by jhuckaby** [comment]
Glad you were able to figure something out.  I will definitely add some kind of persistent option in the next major release.  Thanks!

**Comment by Jacketbg** [comment]
That's very good news! Thank you. In the mean time I've changed it a bit. It turned out that with the Chain reaction setting, the jobs become impossible to be stopped even manually. Even if the event is disabled, aborting the last job triggers the failure chain reaction. Do you want me to report this as a separate bug? So I stopped using Chain reaction and I now use retries, which led to #191 :).
My understanding of persistent jobs is that they should be always running unless manually aborted. Even if they exit with `0` they should be restarted. Some statistics may not be very appropriate for persistent jobs (e.g. runtime) too. I hope this helps you improve this project 👍

**Comment by dossy** [comment]
Indeed, jobs that are scheduled for every minute (i.e., `* * * * *`) with concurrency=1 need special treatment - I don't want notifications for `Maximum of 1 job already running for event` errors, but I do want all other error notifications.

Maybe there could be a simple checkbox in the schedule for each job in the "Misc. Options" section that is unchecked by default, and when checked, is: "Don't send notifications when schedule would exceed concurrency" - in case there's a useful use case for this for other schedules, I wouldn't make this behavior automatic based on the schedule/concurrency settings, but just being able to turn off that notification would be super useful.

I'd be happy to work on this change the approach is acceptable and submit a PR, because I'm going to be scheduling a lot of jobs using Cronicle and I have error notifications going to a Slack webhook, and the "max of 1 job" error spam is a problem for me.

**Comment by jhuckaby** [comment]
@dossy Try setting your event timing to "On Demand", then set the "Chain Reaction" system to run itself on completion.  Meaning, select the same event to run as a chain reaction.  Then fire it off once manually using the "Run Now" button, and it'll run continuously forever, with no errors or notifications.

**Comment by jhuckaby** [comment]
> Even if the event is disabled, aborting the last job triggers the failure chain reaction. Do you want me to report this as a separate bug?

@Jacketbg Yes, this is a separate bug.  I will add this to the TODO list.  Thanks!

**Comment by jhuckaby** [comment]
> Even if the event is disabled, aborting the last job triggers the failure chain reaction. Do you want me to report this as a separate bug?

This has been fixed in [v0.8.30](https://github.com/jhuckaby/Cronicle/releases/tag/v0.8.30).  Thanks for reporting it!

You should now be able to achieve a continuous running job by chaining an event to itself, even if you use the chain on failure.  Aborting jobs manually will no longer fire a chain reaction.

Thanks!

**Comment by jhuckaby** [comment]
Added a wiki to explain how to setup a continuously running job with On Demand timing and the Chain Reaction system: https://github.com/jhuckaby/Cronicle/wiki/Continuously-Running-Job

**Comment by dossy** [comment]
@jhuckaby - Will that survive server process restarts, or will I have to manually kick it off each time?

Also: What if the job typically only takes ~15 seconds to run, and I don't actually want it to start immediately after finishing?  I want it to start at most once a minute, but I just don't want the error notification if any individual job execution prevents the next scheduled job from starting.

**Comment by jhuckaby** [comment]
> Will that survive server process restarts, or will I have to manually kick it off each time?

If you restart the **master** server, then you'll have to manually start the job again.  But that should be a rare event indeed.  Cronicle is designed to run continuously 24x7.  I'll have a better way to accomplish this in the next major release.

> Also: What if the job typically only takes ~15 seconds to run, and I don't actually want it to start immediately after finishing? I want it to start at most once a minute

The best way to achieve this right now is to use the Shell Plugin, and add a `sleep` statement after you run your command, e.g. `sleep 60`.  This will insure the job takes at least 60 seconds to complete before running the chain reaction, even if your command exits instantly.

**Comment by Jacketbg** [comment]
That's great news, Thank You! I will update my instance. I got used to using retries as a means to keep persistent tasks, because of this bug. I needed to add `exit 1` at the end of each script to be sure that the jobs are always retried. With chaining it's probably a good idea to have at least `sleep 1` at the end of the script.
